lib/syskit/cli/log_runtime_archive.rb

```diff
@@ -41,6 +42,42 @@ def process_root_folder
                 end
             end
 
+            # Manages folder available space
+            #
+            # The method will check if there is enough space to save more log files
+            # according to pre-established threshold.
+            #
+            # @param [integer] free_space_low_limit: required free space threshold, at
+            #   which the archiver starts deleting the oldest log files
+            # @param [integer] free_space_delete_until: post-deletion free space, at which
+            #   the archiver stops deleting the oldest log files
+            def ensure_free_space(free_space_low_limit, free_space_delete_until)
+                if free_space_low_limit > free_space_delete_until
+                    raise ArgumentError,
+                          "cannot erase files: freed limit is smaller than " \
+                          "low limit space."
+                end
+
+                stat = Sys::Filesystem.stat(@target_dir)
+                available_space = stat.bytes_free
+
+                return if available_space > free_space_low_limit
+
+                until available_space >= free_space_delete_until
+                    files = @target_dir.each_child.select(&:file?)
+                    if files.empty?
+                        Roby.warn "Cannot erase files: the folder is empty but the "\
+                        "available space is smaller than the threshold."
+                        break
+                    end
+
+                    removed_file = files.min
```

Comment:
Right ... I am wrong w.r.t. the fact that we can sort the filenames alphabetically.

The file names currently are `<basename>.<number>.tar`, where `number` is not normalized, i.e. `1`, `2`, `3`, `4`... Unfortunately, this means that `somefile.40.log` is lower than `somefile.5.log`, which is rather obviously not what we want.

However, I think the best would be to normalize the archive names to be prefixed with zeroes (e.g. `00040`, `00005`. So, let's keep this PR as-is and do the change in a further PR. We need to have (1) a disk full and (2) more than 100G of logs for that bug to become an issue.... I can live with that.